### PR TITLE
Add factory fixture for deploymentconfig pod

### PR DESCRIPTION
**Pull request by prsurve**
1- Add factory fixture for deployment config pod
2- Move script from fixture to factory fixture

Signed-off-by: prsurve <prsurve@redhat.com>

### CODE REVIEW

**Comment by prsurve** [comment]
[Logs](https://github.com/red-hat-storage/ocs-ci/files/3551181/factory_pod_logs.log)

File: tests/fixtures.py [hunk 1]
```diff
@@ -298,7 +298,8 @@ def finalizer():
     class_instance.dc_pod_objs = [
         helpers.create_pod(
             interface_type=class_instance.interface, pvc_name=pvc_obj.name, do_reload=False,
```
Comment (ebenahar) [inline tests/fixtures.py]: do we still need this fixture? maybe it would be better to use the new factory fixture for deployment config

Comment (prsurve) [reply tests/fixtures.py]: This is an old fixture if someone needs they can use this but for my script, I am using factory fixture

File: tests/conftest.py [hunk 2]
```diff
@@ -429,6 +429,112 @@ def finalizer():
...
+                by using these data. Parameter `pvc` is not used but reference
+                is set if provided.
+            replica_count (int): Replica count for deployment config
+        """
+        if custom_data:
+            dc_pod_obj = helpers.create_resource(**custom_data)
+        else:
+
+            pvc = pvc or pvc_factory(interface=interface, size=size)
+            sa_obj = service_account_factory(project=pvc.project)
```
Comment (fbalak) [inline tests/conftest.py]: User should have option to add service_account via parameter (as with pvc)

File: tests/conftest.py [hunk 3]
```diff
@@ -429,6 +429,112 @@ def finalizer():
...
+            )
+            instance.delete()
+            instance.ocp.wait_for_delete(resource_name=instance.name)
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
+@pytest.fixture()
+def dc_pod_factory(
```
Comment (fbalak) [inline tests/conftest.py]: Would it make sense to enhance `pod_factory` with new parameters like `service_account` and `dc_deployment` instead of creating this new fixture?

File: tests/conftest.py (hunk 2 above)
Comment (prsurve) [reply tests/conftest.py]: @fbalak sorry i didn't get you

File: tests/conftest.py (hunk 3 above)
Comment (prsurve) [reply tests/conftest.py]: personally, I feel we should have separate dc_pod_factory it will be easy for everyone to use if we suppose to make a change in pod_factory users  need to add additional data to pvc_factory for setting up dc_pods

File: tests/conftest.py (hunk 2 above)
Comment (fbalak) [reply tests/conftest.py]: I am sorry that I did not make myself clear. I meant to add `service_account` to factory function like:
```
   def factory(
        interface=constants.CEPHBLOCKPOOL,
        pvc=None,
        service_account=None,
        size=None,
        custom_data=None,
        replica_count=1,
    ):
```
and then fill this variable like:
```
sa_obj = service_account or service_account_factory(project=pvc.project)
```
This would allow user that uses `dc_pod_factory` to control what service account is used in created pod object.

Comment (prsurve) [reply tests/conftest.py]: @fbalak service_account should be created under newly created namespace be cannot add service account from other namespace dc pod won't come in running state

File: tests/conftest.py [hunk 4]
```diff
@@ -429,6 +429,112 @@ def finalizer():
...
+        """
+        Args:
+            project (object): ocs_ci.ocs.resources.ocs.OCS instance
+                of 'Project' kind.
+
+        Returns:
+            object: helpers.create_serviceaccount instance.
+        """
+        nonlocal active_service_account_obj
+        if active_service_account_obj:
```
[the remark this answers is not in the document]
Comment (vasukulkarni) [reply tests/conftest.py]: please ignore, its initialized in closure function second time

File: tests/conftest.py (hunk 2 above)
Comment (fbalak) [reply tests/conftest.py]: Right, but if you add the parameter then you will have option to test adding service_account from different namespace it as negative testcase or create two pods with different user accounts in same namespace.

Comment (prsurve) [reply tests/conftest.py]: Done

File: tests/manage/pv_services/test_run_io_multiple_dc_pods.py [hunk 5]
```diff
@@ -1,41 +1,24 @@
...
+        pytest.param(
+            constants.CEPHFILESYSTEM, marks=pytest.mark.polarion_id("OCS-1285")
+        )
+    ]
 )
-class BaseRunIOMultipleDcPods(ManageTest):
+class TestRunIOMultipleDcPods(ManageTest):
     """
-    Run IO on multiple dc pods in parallel
+    Run IO on multiple dc pods in parallel - RBD
```
Comment (RazTamir) [inline tests/manage/pv_services/test_run_io_multiple_dc_pods.py]: just RBD?

Comment (prsurve) [reply tests/manage/pv_services/test_run_io_multiple_dc_pods.py]: Rbd and ceph fs both will update doc string

**Comment by ebenahar** [comment]
@prsurve Following new commits, please re-verify

**Comment by prsurve** [comment]
Logs:- https://paste.fedoraproject.org/paste/z6xgOQIue5jgqcrlCnHXCg
